### PR TITLE
Fixes for vertex fits in the muon system

### PR DESCRIPTION
**Pull request by steggema**
#### PR description:

Problem addressed: In EXO long-lived analyses, it was observed that using CMSSW vertex fitting code to reconstruct the decay vertex of resonances decaying beyond around 300 cm had much worse efficiency than for lower values, and the covariance matrix entries remained at their default values.

This was identified as coming from an issue in the fitting code when the sign of the magnetic field switches sign between two extrapolated track states. A report given in the context of the related Muon POG studies can be found here: https://indico.cern.ch/event/1220371/#6-dsa-vertex-inefficiency-stud
Another report will be given at a tracking POG meeting: https://indico.cern.ch/event/1232392/

The PR also includes a minimal change to the SequentialVertexFitter such that user code can update the boundaries in which the fit is considered valid. Up to now, users were updating these hard-coded values manually. We leave more involved updates (like taking the default values from a database or making it configurable at the EDProducer level) to future studies.

Changes to output: No changes to outputs are expected since this PR only affects the special use case of searches for long-lived particles using the CMSSW vertex fitting code to reconstruct secondary vertices outside of the inner tracker.

Other PRs: There is no interference with other PRs that we are aware of.

#### PR validation:

The proposed fixes lead to a large increase of efficiency for vertex fits beyond around 300 cm, and also to reasonable values of the covariance matrix entries, as reported in the slides above.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport.

@annamasce @jalimena @aescalante

### CODE REVIEW

**Comment by mmusich** [comment]
type tracking

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mandrenguyen** [comment]
assign tracking-pog
A priori looks reasonable, but let's see if anything comes up in the tracking POG presentation on Monday.

File: RecoVertex/VertexTools/interface/SequentialVertexFitter.h [hunk 1]
```diff
@@ -226,6 +242,9 @@ class SequentialVertexFitter : public VertexFitter<N> {
   float theMaxShift;
   int theMaxStep;
 
+  float trackerBoundsRadius{112.};
```
Comment (mmusich) [inline RecoVertex/VertexTools/interface/SequentialVertexFitter.h:245]: if you don't plan to make it configurable (as per PR description), at least preserve the original comment about having to take it from elsewhere.

Comment (mmusich) [reply RecoVertex/VertexTools/interface/SequentialVertexFitter.h:246]: addressed at https://github.com/cms-sw/cmssw/pull/40479/commits/170b1938f837ef274be19a942ddccac4a83c89d0

**Comment by mmusich** [comment]
+1
   * based on the discussion at the [TRK POG meeting of Jan 2019](https://indico.cern.ch/event/1232392/?note=222231)

**Comment by mandrenguyen** [comment]
+reconstruction
No changes to comparisons, but this addresses a rare case, as explained in the PR description.
Code changes are in line with description.

**Comment by perrotta** [comment]
+1

**Pull request merged**